src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs

```diff
@@ -271,12 +271,20 @@ private void WriteStringValue(DateTimeOffset value, ref int idx)
 
         private void FormatLoop(DateTimeOffset value, ref int idx)
         {
-            if (!Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out int bytesWritten, s_dateTimeStandardFormat))
+            Span<byte> tempSpan = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            bool result = Utf8Formatter.TryFormat(value, tempSpan, out int bytesWritten, s_dateTimeStandardFormat);
+            Debug.Assert(result);
+
+            JsonWriterHelper.TrimDateTimeOffset(tempSpan.Slice(0, bytesWritten), out bytesWritten);
+
+            if (_buffer.Length < bytesWritten)
```

Comment:
You forgot to subtract `idx`.

This is the likely cause of the test failures.
[System.Text.Json.Tests.Utf8JsonWriterTests/WriteDateTimeOffsetsValue(formatted: False, skipValidation: False, keyString: \">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>)](https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F35966~2Fmerge/test~2Ffunctional~2Fcli~2F/20190401.9/workItem/System.Text.Json.Tests/analysis/xunit/System.Text.Json.Tests.Utf8JsonWriterTests~2FWriteDateTimeOffsetsValue(formatted:%20False,%20skipValidation:%20False,%20keyString:%20%5C%22%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E%3E))

```text
Unhandled Exception of Type System.ArgumentException
Message :
System.ArgumentException : Destination is too short.
Parameter name: destination
Stack Trace :
   at System.Span`1.CopyTo(Span`1 destination) in /__w/5/s/src/System.Private.CoreLib/shared/System/Span.Fast.cs:line 278
   at System.Text.Json.Utf8JsonWriter.FormatLoop(DateTimeOffset value, Int32& idx) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 288
   at System.Text.Json.Utf8JsonWriter.WriteStringValue(DateTimeOffset value, Int32& idx) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 265
   at System.Text.Json.Utf8JsonWriter.WriteStringMinimized(ReadOnlySpan`1 escapedPropertyName, DateTimeOffset value) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 225
   at System.Text.Json.Utf8JsonWriter.WriteStringByOptions(ReadOnlySpan`1 propertyName, DateTimeOffset value) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 204
   at System.Text.Json.Utf8JsonWriter.WriteStringEscapeProperty(ReadOnlySpan`1 propertyName, DateTimeOffset value, Int32 firstEscapeIndexProp) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 154
   at System.Text.Json.Utf8JsonWriter.WriteStringEscape(ReadOnlySpan`1 propertyName, DateTimeOffset value) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 109
   at System.Text.Json.Utf8JsonWriter.WriteString(ReadOnlySpan`1 propertyName, DateTimeOffset value, Boolean escape) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 52
   at System.Text.Json.Utf8JsonWriter.WriteString(String propertyName, DateTimeOffset value, Boolean escape) in /__w/1/s/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs:line 29
   at System.Text.Json.Tests.Utf8JsonWriterTests.WriteDateTimeOffsetsValue(Boolean formatted, Boolean skipValidation, String keyString) in /__w/1/s/src/System.Text.Json/tests/Utf8JsonWriterTests.cs:line 2881
```